core/src/main/java/org/apache/calcite/rel/metadata/RelMdRowCount.java

```diff
@@ -108,7 +108,11 @@ public Double getRowCount(Intersect rel, RelMetadataQuery mq) {
         rowCount = partialRowCount;
       }
     }
-    return rowCount;
+    if (!rel.all) {
+      return rowCount;
+    } else {
+      return rowCount * 2;
```

Comment:
IMO, it should not return row count which is bigger than the max row count of inputs.

Reply:
`Intersect all` may contain duplicate data. So it should return rowCount * 2. 
`rowCount` depends on the smallest rowCount of the input nodes.

Reply:
@chunwei Is there any problems here ?